### DISCUSSION TITLE
Is there any way to use untyped AST?

### DISCUSSION THREAD

**Discussion by scf37** [scala/scala3#12539] [Metaprogramming] [answered]
I.e. `q"myVal.myField"` alternative when I do not know exact types or have access to definition AST.

### My use case
Given class `C` and type `T`, find all values of `T` referenced in in `C` methods.
### Scala2 solution
parse scala source file, extract string expressions, generate untyped AST:
```scala
  val left = c.universe.TermName(leftStr)
  val right = c.universe.TermName(rightStr)
  q"$left.$right"
```
Then generate seq of all expressions and filter them at runtime
```scala
      q"""
        Seq(..$expressions)
        .filter{v: Any => v.isInstanceOf[T]}
        .asInstanceOf[Seq[T]]"""
```

### Scala3 ?
- Is there any way to get access to full AST of call site class?
- how to generate Expr referring to TermName of unknown type?

**Comment by giiita** [comment, marked as answer]
This question is probably related to this one.

https://github.com/lampepfl/dotty/discussions/12500

However, it is possible to refer to any expression in call site, as in the example below.
It seems that it is no longer possible to refer to it visually from within `Quasiquotes` as before.

```scala
object Main extends App {
  val member: String = "I am Main member"
  println(Macro.getMember())
}
```

```scala
import scala.quoted._
object Macro {
  def getMember()(using q: Quotes): Expr[String] = {
    Select.unique(This(q.reflect.Symbol.spliceOwner.owner), "member").asExprOf[String]
  }
}
```
